plugins/modules/idrac_server_config_profile.py

```diff
@@ -923,46 +1145,61 @@ def main():
         argument_spec=specs,
         required_if=[
             ["command", "export", ["share_name"]],
+            ["command", "export_custom_defaults", ["share_name"]],
             ["proxy_support", True, ["proxy_server"]]
         ],
         supports_check_mode=True)
 
     validate_input(module, module.params.get("scp_components"))
     try:
         http_share = False
+        command = module.params.get("command")
         if module.params.get("share_name") is not None:
             http_share = module.params["share_name"].lower().startswith(('http://', 'https://'))
         with iDRACRedfishAPI(module.params) as idrac:
-            validate_scp_components(module, idrac)
-            command = module.params['command']
-            if command == 'import':
-                command_obj = ImportCommand(idrac, http_share, module)
-            elif command == 'export':
-                command_obj = ExportCommand(idrac, http_share, module)
+            if command in ["import", "export", "preview"]:
+                validate_scp_components(module, idrac)
+            command_map = {
+                "import": (ImportCommand, True),
+                "export": (ExportCommand, True),
+                "import_custom_defaults": (ImportCustomDefaultCommand, False),
+                "export_custom_defaults": (ExportCustomDefaultCommand, False),
+                "preview": (PreviewCommand, True)
+            }
+            command_obj_class, http_share_req = command_map.get(command)
+            if http_share_req:
+                command_obj = command_obj_class(idrac, http_share, module)
             else:
-                command_obj = PreviewCommand(idrac, http_share, module)
+                command_obj = command_obj_class(idrac, module)
             scp_status, changed = command_obj.execute()
 
         if module.params.get('job_wait'):
-            scp_status = strip_substr_dict(scp_status)
+            if command != "export_custom_defaults":
```

Comment:
Try to avoid the if statement in the negative sense, correct this. makes it hard to understand. apply from 1177 to 1188

Reply:
Updated